cmd/app/main.go

```diff
@@ -60,11 +64,38 @@ func main() {
 		log.Fatal(err)
 	}
 
-	server := virtual.NewServer(reg, environment)
+	var server Server = virtual.NewServer(reg, environment)
 
 	log.Printf("listening on port: %d\n", *port)
 
+	go func(server Server) {
+		waitForSignal()
+		shutdown(server, *shutdownTimeout)
+	}(server)
+
 	if err := server.Start(*port); err != nil {
+		shutdown(server, *shutdownTimeout)
```

Comment:
I see. I don't return the error because either the error from `err := server.Start(*port)`  or shutdown() must be logged with `Error`, while the other is logged with `Fatal`. We cannot do both unless we use multiple errors wrapping from Go 1.20, but I don't think in this case makes sense.